src/data/bucket/fill_extrusion_bucket.js

```diff
@@ -301,7 +301,7 @@ class FillExtrusionBucket implements Bucket {
 
     addFeature(feature: BucketFeature, geometry: Array<Array<Point>>, index: number, canonical: CanonicalTileID, imagePositions: {[_: string]: ImagePosition}) {
         const flatRoof = this.enableTerrain && feature.properties && feature.properties.hasOwnProperty('type') &&
```

Comment:
I suppose we don't need the `type` check in `properties` as well? It's different from `feature.type` and also a data property we don't want to rely on.

```suggestion
        const flatRoof = this.enableTerrain && feature.properties &&
```